zerver/actions/message_send.py

```diff
@@ -261,18 +261,11 @@ def should_send(setting: str, row: Dict[str, Any]) -> bool:
             if should_send("email_notifications", row)
         }
 
-        if possible_wildcard_mention:
-            # If there's a possible wildcard mention, we need to
-            # determine the set of users who have enabled the
-            # "wildcard_mentions_notify" setting (that is, the set of
-            # users for whom wildcard mentions should be treated like
-            # personal mentions for notifications). This setting
-            # applies to both email and push notifications.
-            wildcard_mention_user_ids = {
-                row["user_profile_id"]
-                for row in subscription_rows
-                if should_send("wildcard_mentions_notify", row)
-            }
+        wildcard_mention_user_ids = {
+            row["user_profile_id"]
+            for row in subscription_rows
+            if should_send("wildcard_mentions_notify", row)
+        }
```

Comment:
So I think there's actually a reason not to do this -- for a standard message to a large stream with 20K recipients, we'd be sending this very large data structure of 20K user IDs through the events system for no reason. I could remember wrong, but I think we've been pretty careful to avoid sending extra copies of "Every user ID on the stream".

I think a better option would be to just have an `else: wildcard_mention_user_ids = []` here if you want to rely on the field always being present?

Definitely should be documented in a comment if so.

Reply:
I think this approach is compatible with your second commit's concept, though.

The PR otherwise looks good to me.